skops/card/_model_card.py

```diff
@@ -1,27 +1,54 @@
 from __future__ import annotations
 
-import copy
 import json
 import re
-import shutil
-import tempfile
-from dataclasses import dataclass
+import sys
+import textwrap
+from dataclasses import dataclass, field
 from pathlib import Path
 from reprlib import Repr
-from typing import Any, Optional, Union
+from typing import Any, Iterator, Sequence, Union
 
-from huggingface_hub import CardData, ModelCard
+from huggingface_hub import CardData
 from sklearn.utils import estimator_html_repr
 from tabulate import tabulate  # type: ignore
 
-import skops
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
```

Comment:
The problem when putting this into `fixes.py` and importing it is that for some reason, mypy doesn't understand the protocol anymore. E.g. it'll claim:

> skops/card/_model_card.py:694: error: Argument 2 to "_add_single" of "Card" has incompatible type "PlotSection"; expected "Union[Formattable, str]"

even though `PlotSection` conforms to the `Formattable` protocol. Now we could put a `# type: ignore` on this method, but then we might as well not use a protocol at all, if it is never checked.

Yes, it is annoying that this doesn't work. However, in this case I'd be in favor of keeping it as is because it's a temporary fix anyway. As soon as we can stop supporting Python 3.7, all these gymnastics can go away (official EOL for Python 3.7 is 27 June 2023).

Reply:
ok, then we can add a comment here and in `fixes.py` to remember it.